src/repository/conUOpenDataCourseRepository.js

```diff
@@ -0,0 +1,120 @@
+     */
+    static create(data) {
+        return new Promise((resolve, reject) => {
+            const newOpenDataCourse = createOpenDataCourse(data)
+            newOpenDataCourse.save((err, faculty) => {
+                if (err) {
+                    reject(err);
+                }
+                resolve(faculty);
+            })
+        })
+    }
+
+    /**
+     * Find all courses with a specific course code.
+     * @param {string} courseCode, the course code of interest.
+     * @returns [{Course}] Returns a promise. Resolves with an array of Courses associated with the courseCode param.
+     */
+    static findAllByCourseCode(courseCode) {
+        return new Promise((resolve, reject) => {
+            OpenDataCourse.find({
+                subject: {"$in": [courseCode.toUpperCase()]}
+            })
+                .then((result) => {
+                    resolve(result);
+                })
+                .catch(err => reject(err))
+        })
+    }
+
+    /**
+     * Find a specific course with a specific course code and course number.
+     * @param {string} courseCode the course code of interest.
+     * @param {string, int} courseNumber the course number of interest.
+     * @returns [{Course}] Returns a promise. Resolves with an array of Course associated with the courseCode and courseNumber param.
+     */
+    static findByCourseCodeAndNumber(courseCode, courseNumber) {
+        return new Promise((resolve, reject) => {
+            OpenDataCourse.find({
+                subject: {"$in": [courseCode.toUpperCase()]},
+                catalog: {"$in": [courseNumber.toString()]}
+            })
+                .then((result) => {
+                    resolve(result);
+                })
+                .catch(err => reject(err))
+        })
+    }
+
+    /**
+     * Drops the opendatacourses table
+     * @returns boolean, returns true if the table is dropped, returns false if the table is not dropped
+     */
+    static dropTable() {
+        return new Promise((resolve, reject) => {
+            OpenDataCourse.collection.drop().then((result) => {
+                resolve(result);
+            }).catch((err) => {
+                reject(err);
+            })
+        })
+    }
+
+    /**
+     * Creates table of Course records using an array of Course records.
+     * @returns [{Course}] Returns a promise. Resolves with an array of Courses added to the table.
+     */
+    static batchCreateCourse(coursesData) {
+        return new Promise((resolve, reject) => {
+            OpenDataCourse.collection.insertMany(coursesData, (err, docs) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(docs);
+                }
+            })
+        })
+    }
+
+    /**
+     * Refreshes Course data in the openadatacourses table using the course data in Concordia's Open Data
+     */
+    static refreshCourseData(){
```

Comment:
While there is no problem in the code, this should not be in the repository file. The repository file should only contain functions that define transactions with our MongoDB database. These functions should be moved to a helper file.